### ISSUE TITLE
`start()`/`stop()` doesn't cancel the previous promise, it will be called at the next `start()`(when is finished)

### ISSUE THREAD

**Issue by lopi-py** [chriscerie/roact-spring#21]
I set `enabled` to `false` on `MouseEnter` event and `true` on `MouseLeave` event. When I hover, it plays the 1.3 scale animation and then prints "finished", all good but the issue comes when I quickly(before 1.3 scale animation finishes) move the cursor out of the frame, it plays the 1 scale animation immediately, all good so far, **but** `finished` still prints, it should not print, it should cancel, if I call `motor.stop()` before 1 scale animation it still doesn't work, is this intended or is there a workaround?. I noticed that when 1 scale animation finishes, "finished" is printed
![image](https://user-images.githubusercontent.com/70210066/167324446-deea8e3b-5219-4922-86cf-7c59b04c5029.png)

**Comment by chriscerie** [comment]
This behavior is consistent with react-spring's `onRest` prop during imperative updates, where promises from subsequent `start`s get stalled and resolved at the same time once the spring stops moving.

Ideally they'd resolve with an `AnimationResult` table so you can check if the spring was cancelled or interrupted with another `start` to achieve the behavior you want.

**Comment by lopi-py** [comment]
It's not called on `stop` or `play` immediately, it seems to be making pending the promise, here are the cycles:

#### case 1 (fine)
`start({ scale = 1.3 })` -> finishes(**1.3**) -> **print**("finished")

#### case 2 (issue)
`start({ scale = 1.3 })` -> `play({ scale = 1 })`(1.3 does not print "finished") -> finished(**1**) -> **print**("finished")

It's annoying for me because I am making that on `MouseEnter` the frame grows and(`using andThen`) make it not visible but on `MouseLeave`, the frame returns back to the original scale(1) but when scale(1) finishes the previous promise that makes grow the frame get called(when scale 1 finishes) and makes it not visible :/

**Comment by chriscerie** [comment]
Yes, this is intended behavior (see the first reply). Your issue would be solved from the `AnimationResult` table I mentioned so you would be able to check if the spring was interrupted and only move forward if it wasn’t.

As a hacky workaround you can check if the value of `scale` is 1.3 in the `andThen` call. If it is not 1.3, then you’d know the spring was interrupted by the other `start` call.

```lua
motor.start({ scale = 1.3 }):andThen(function()
    if scale:getValue() == 1.3 then
        print("Finished")
    end
end)
```

**Comment by lopi-py** [comment]
Ah alright, thanks
